application/src/main/java/org/opentripplanner/routing/algorithm/filterchain/filters/transit/RemoveTransitIfStreetOnlyIsBetter.java

```diff
@@ -14,10 +15,34 @@
  */
 public class RemoveTransitIfStreetOnlyIsBetter implements RemoveItineraryFlagger {
 
+  private static final Consumer<RemoveTransitIfStreetOnlyIsBetterResults> IGNORE_SUBSCRIBER =
+    i -> {};
+
   private final CostLinearFunction costLimitFunction;
+  private final Cost generalizedCostMaxLimit;
+  private final Consumer<
+    RemoveTransitIfStreetOnlyIsBetterResults
+  > removeTransitIfStreetOnlyIsBetterResultsSubscriber;
 
-  public RemoveTransitIfStreetOnlyIsBetter(CostLinearFunction costLimitFunction) {
+  /**
+   * Constructs the RemoveTransitIfStreetOnlyIsBetter filter.
+   * @param costLimitFunction the cost limit function to use with the filter
+   * @param generalizedCostMaxLimit this limit is not null when paging is used
+   * @param removeTransitIfStreetOnlyIsBetterResultsSubscriber this subscriber stores the generalizedCostMaxLimit for use with paging
+   */
+  public RemoveTransitIfStreetOnlyIsBetter(
+    CostLinearFunction costLimitFunction,
+    Cost generalizedCostMaxLimit,
+    Consumer<
+      RemoveTransitIfStreetOnlyIsBetterResults
+    > removeTransitIfStreetOnlyIsBetterResultsSubscriber
+  ) {
     this.costLimitFunction = costLimitFunction;
+    this.generalizedCostMaxLimit = generalizedCostMaxLimit;
+    this.removeTransitIfStreetOnlyIsBetterResultsSubscriber =
+      removeTransitIfStreetOnlyIsBetterResultsSubscriber == null
+        ? IGNORE_SUBSCRIBER
+        : removeTransitIfStreetOnlyIsBetterResultsSubscriber;
```

Comment:
Perhaps it would be better to have to constructors, without and without this consumer or at least the param should be marked as `@Nullable` on the constructor. However, I'm not sure if I'm a fan of this IGNORE_SUBSCRIBER. It would be more explicit just to check if the subscriber is specified before running the accept and put some documentation around it telling when it's specified.

Reply:
I removed the IGNORE_SUBSCRIBER here and in NumItinerariesFilter. I check the value for null in the constructors

Reply:
For the record, I think these to approches are equally good. Using a DO_NOTHING lamda simplifies the main logic, but increase the complexity of the setup.